.github/workflows/minor_version_release.yml

```diff
@@ -25,17 +21,47 @@ jobs:
           python -m pip install --upgrade pip
           pip install pydoc-markdown==4.5.1
 
-      - name: Convert version name to regular semver (e.g. v1.10.x -> v1.10.0)
-        id: regular_semver
-        env:
-          version: ${{github.event.inputs.version}}
+      - name: Get current version
+        id: current-version
+        shell: bash
+        # We only need `major.minor` in Readme so we cut the full version string to the first two tokens
+        run: echo "::set-output name=minor::$(cat VERSION.txt | cut -d "." -f 1,2)"
+
+      - name: Create new version branch
+        run: |
+          git checkout -b v${{ steps.current-version.outputs.minor }}.x
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git push -u origin v${{ steps.current-version.outputs.minor }}.x
+
+      - name: Bump version on main
+        id: bump-version
+        run: |
+          git fetch
+          git checkout main
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git pull
+          python ./.github/utils/bump_version.py
+          echo "::set-output name=new_minor::$(cat VERSION.txt | cut -d "." -f 1,2)"
+          git checkout -b bump-version
+          git add ./VERSION.txt
+          git commit -m "Bump version"
+          git push -u origin bump-version
+
+      # Note that patch versions all sync to the one readme minor version
+      # e.g. Haystack 1.9.1 and 1.9.2 both map to Readme 1.9
+      - name: Release Readme version
         run: |
-            echo $version | sed -e 's/x$//' | { var=$(cat); echo "VERSION=v${var}0" >> $GITHUB_ENV; }
-            echo $version | sed -e 's/x$//'
+          git checkout ${{github.event.inputs.version}}
+          python ./.github/utils/release_docs.py --version v${{ steps.current-version.outputs.minor }} --key ${{ secrets.README_API_KEY }}
```

Comment:
If I got the python script right, this should be the "new minor"? In case this should be:
```
--version v${{ steps.bump-version.outputs.minor }}
```

Reply:
The original is actually correct but I realize it is terminologically quite confusing. There are technically 4 version names that we are working with.

old-release-version: 1.9
old-unstable-version: 1.10
new-release-version: 1.10
new-unstable-version: 1.11

For clarity, I only make reference to `steps.versions.outputs.new_release_version` in my latest changes